### PR TITLE
lecture 18 scribe

### PR DESCRIPTION
**Pull request by yueatsprograms**

### CODE REVIEW

File: notes/lecture18.tex [hunk 1]
```diff
@@ -0,0 +1,146 @@
+\section{Lecture 18: Saddle points} 
+
```
Comment (mrtzh) [inline notes/lecture18.tex]: Please start with short summary.

File: notes/lecture18.tex [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+\begin{definition}
+$x^*$ is a stationary point if the gradient vanishes at $x^*$ i.e. $\nabla f(x^*) = 0$.
```
Comment (mrtzh) [inline notes/lecture18.tex]: Don't start sentence with symbol.

File: notes/lecture18.tex [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+$f(x)\leq f(x^*)\leq f(y)$.
+\end{definition}
+
+\begin{definition}
+For $f\in C^2$, a saddle point $x^*$ is a strict saddle point if the Hessian at that point is not positive semidefinite, i.e.
+$\lambda_{\text{min}}(\nabla^2 f(x^*)) < 0$,
+where $\lambda_{\text{min}}$ denotes the smallest eigenvalue.
+\end{definition}
+
+\noindent
```
Comment (mrtzh) [inline notes/lecture18.tex]: No need to add \noindent here or elsewhere.

File: notes/lecture18.tex [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+This lecture formalizes and proves the following intuitive statement for nonconvex optimization:
+\begin{center}
+\textbf{Gradient descent almost never converges to strict saddle points.}
+\end{center}
+as presented in \textit{Gradient Descent Converges to Minimizers} by Jason D. Lee, Max Simchowitz, Michael I. Jordan, and Benjamin Recht.
```
Comment (mrtzh) [inline notes/lecture18.tex]: Please add proper bib entry.

File: notes/lecture18.tex [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+\noindent
+Let's start by viewing each gradient descent update as a operator.  
+For a fixed step size $\eta$, define
+$$
+g(x) = x-\eta\nabla f(x)
+$$ 
+so the notation for the result of iteration $t$ from our previous discussion of gradient descent carries over as $x_t = g^t(x_0) = g(g(...g(x_0)))$, where $g$ is applied $t$ times on the initial point $x_0$. $g$ is called the gradient map. Note that $x^*$ is stationary iff. it is a fixed point of the gradient map i.e. $g(x^*) = x^*$. Also note that
```
Comment (mrtzh) [inline notes/lecture18.tex]: Don't start sentence with symbol (second sentence here).

File: notes/lecture18.tex [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+\end{definition}
+
+\begin{definition}
+For $f\in C^2$, a saddle point $x^*$ is a strict saddle point if the Hessian at that point is not positive semidefinite, i.e.
+$\lambda_{\text{min}}(\nabla^2 f(x^*)) < 0$,
+where $\lambda_{\text{min}}$ denotes the smallest eigenvalue.
+\end{definition}
+
+\noindent
+This lecture formalizes and proves the following intuitive statement for nonconvex optimization:
```
Comment (mrtzh) [inline notes/lecture18.tex]: The lecture can't prove.

File: notes/lecture18.tex [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+(1+\eta)y - \eta y^3 
+\end{bmatrix}
+$, and
+$\nabla^2 f(x,y) = 
+ \begin{bmatrix}
+1 & 0 \\
+0 & 3y^2-1 \\
+\end{bmatrix}
+$.
+So $(0,-1)$ and $(0,1)$ are the local minima, and $(0,0)$ is the only strict saddle point. Similar to in the previous example,
```
Comment (mrtzh) [inline notes/lecture18.tex]: Incomplete sentence at the end.

File: notes/lecture18.tex [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+\end{remark}
+
+\begin{remark}
+By definition, this also holds true to any probability measure absolutely continuous w.r.t. the Lebesgue measure (e.g. any continuous probability distribution).
+\end{remark}
+
+\noindent
+However, the theorem above is only an asymptotic statement. Nonasymptotically, even with fairly natural random initialization schemes and non-pathological functions, gradient descent can be significantly slowed down by saddle points. 
+The most recent result is that gradient descent takes exponential time to escape saddle points (even though the theorem above says that they do escape eventually), as presented in the paper
+\textit{Gradient Descent Can Take Exponential Time to Escape Saddle Points} by Simon S. Du, Chi Jin, Jason D. Lee, Michael I. Jordan, Barnabas Poczos and Aarti Singh. We won't cover this result in this lecture.
```
Comment (mrtzh) [inline notes/lecture18.tex]: Please add bib.

**Comment by yueatsprograms** [comment]
Thanks for the careful reading. Made the changes as requested.

**Comment by mrtzh** [comment]
Looks good. Can you just update your local branch real quick. There've been updates since you sent the pull request. Thanks!

**Comment by yueatsprograms** [comment]
Done :)

**Pull request merged**